service/collector/deployment.go

```diff
@@ -90,6 +91,8 @@ func (d *Deployment) Collect(ch chan<- prometheus.Metric) error {
 	}
 	defer watcher.Stop()
 
+	alreadyEmitted := map[string]struct{}{}
```

Comment:
Why using a `map[string]struct{}{}`, it would be simpler to use a `map[string]bool` IMO and assign true when metric is emitted. WDYT ?